beacon_node/beacon_chain/src/schema_change.rs

```
@@ -93,6 +94,47 @@ pub fn migrate_schema<T: BeaconChainTypes>(
 
             Ok(())
         }
+        // Downgrade v5 to v4
+        (SchemaVersion(5), SchemaVersion(4)) => {
+            info!(&db.log, "Dowgrade v5 to v4");
+            if let Some(OnDiskStoreConfig {
+                slots_per_restore_point,
+            }) = db.hot_db.get(&CONFIG_KEY)?
+            {
+                // If this is a checkpoint-sync'd DB then we can't downgrade.
+                match db.load_anchor_info() {
+                    Ok(opt) => {
+                        match opt {
+                            Some(_) => {
+                                // Whoops, contains a load anchor so this DB can't be downgraded
+                                warn!(
+                                    &db.log,
+                                    "Cannot dowgrade v5 to v4 because DB contains checkpoints"
+                                );
+                                return Err(HotColdDBError::UnsupportedSchemaVersion {
+                                    target_version: to,
+                                    current_version: from,
+                                }
+                                .into());
+                            }
+                            None => (),
+                        }
+                    }
+                    Err(_) => (),
```

We should error if the store errors, rather than failing silently, i.e. match on `db.load_anchor_info()?`

---

Done